src/execution_algorithms/ltfb/truncation_selection_exchange.cpp

```diff
@@ -0,0 +1,311 @@
+  std::string buf;
+  buf.resize(size);
+  comm.recv(reinterpret_cast<El::byte*>(buf.data()), size, src_trainer);
+  return buf;
+}
+// Unpack received model
+void unpack(model& m, std::string const& str) 
+{
+  std::istringstream iss(str);
+  {
+    RootedBinaryInputArchive ar(iss, m.get_comm()->get_trainer_grid());
+    ar(m);
+  }
+}
+} // namespace
+
+// TruncationSelectionExchange implementation
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  std::unordered_map<std::string, metric_strategy> metrics, int truncation_k)
+  : m_metrics{std::move(metrics)},m_truncation_k{std::move(truncation_k)}
+{
+  LBANN_ASSERT(m_metrics.size()==1); //only single (1) metric is supported at this time
+}
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  std::string metric_name,
+  metric_strategy winner_strategy,int truncation_k)
+  : TruncationSelectionExchange({{metric_name, winner_strategy}},truncation_k)
+{}
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  TruncationSelectionExchange const& other)
+  : m_metrics{other.m_metrics}, m_truncation_k{other.m_truncation_k}
+{}
+
+EvalType TruncationSelectionExchange::evaluate_model(model& m,
+                                       ExecutionContext& ctxt,
+                                       data_coordinator& dc) const
+{
+  // Make sure data readers finish asynchronous work
+  const auto original_mode = ctxt.get_execution_mode();
+  dc.collect_background_data_fetch(original_mode);
+
+  //Can use validation if it is global
+  if (!dc.is_execution_mode_valid(execution_mode::testing)) {
+    LBANN_ERROR("LTFB truncation selection requires ",
+                to_string(execution_mode::testing),
+                " execution mode");
+  }
+
+  // Mark the data store as loading - Note that this is a temporary fix
+  // for the current use of the tournament
+  m.mark_data_store_explicitly_loading(execution_mode::testing);
+
+  // Evaluate model on test (or validation?) set
+  get_trainer().evaluate(&m, execution_mode::testing);
+
+  // Get metric values
+  bool found_metric = false;
+  EvalType score=0.f;
+  std::string metric_name;
+  for (const auto& met : m.get_metrics()) {
+    metric_name = met->name();
+    if (m_metrics.count(metric_name)) {
+      found_metric = true;
+      score += met->get_mean_value(execution_mode::testing);
+      break;
+    }
+  }
+
+  //sanity check
+  if (!found_metric) {
+    std::stringstream err;
+    err << "could not find metric \"" << metric_name << "\""
+        << "in model \"" << m.get_name() << "\"";
+    LBANN_ERROR(err.str());
+  }
+
+  m.make_data_store_preloaded(execution_mode::testing);
+
+  // Clean up and return metric score
+  m.reset_mode(ctxt, original_mode);
+  dc.reset_mode(ctxt);
+  return score;
+}
+
+void TruncationSelectionExchange::select_next(model& m,
+                                         ltfb::ExecutionContext& ctxt,
+                                         data_coordinator& dc) const
```

Comment:
```suggestion
void TruncationSelectionExchange::select_next(model& m,
                                              ltfb::ExecutionContext& ctxt,
                                              data_coordinator& dc) const
```
Or just run `clang-format`.